packages/app/src/cli/utilities/developer-platform-client/app-management-client.test.ts

```diff
@@ -113,3 +173,82 @@ describe('allowedTemplates', () => {
     expect(got).toEqual([templateWithoutRules, allowedTemplate])
   })
 })
+
+describe('devStoresForOrg', () => {
+  test('returns a list of OrganizationStore typed stores, from query response', async () => {
+    const mockStore1 = STORE1_NODE
+    const mockStore2 = STORE2_NODE
+
+    const mockAppManagementClient = new AppManagementClient()
+    vi.mocked(businessPlatformOrganizationsRequest).mockResolvedValue(LIST_DEV_STORE_RESPONSE_VALUE)
+
+    const got = await mockAppManagementClient.devStoresForOrg(ORG1.id)
```

Comment:
To me it makes sense to test this call directly here, but in this `devStoresForOrg` we call `await this.businessPlatformToken()` to get the token to call with the request. This blocks this test from running bc of 
```
    if (isUnitTest()) {
      throw new Error('AppManagementClient.businessPlatformToken() should not be invoked dynamically in a unit test')
    }
```

I want to understand why this check is here, and if there's a way to can get around it?  I've attempted to mock `ensureAuthenticatedBusinessPlatform` and I can't set `_businessPlatformToken` directly because it's a private field of `AppManagementClient`.

Thoughts? @amcaplan @isaacroldan

Reply:
@amcaplan or @gonzaloriestra have more context on why this is here, but i think we should remove that error.

Mocking `ensureAuthenticatedBusinessPlatform` seems like the right approach

Reply:
> A bit related too, I see we used to test API fetches from Partners [here](https://github.com/Shopify/cli/blob/main/packages/app/src/cli/services/dev/fetch.test.ts#L163) (Unrelated but I think this should be removed cause the corresponding function this is testing doesn't seem to be used anywhere any more). Is there a place where we documented the decision to rely on type assertions when it came to these new query calls? Either way, I think CLI should be consistent in the ways we test queries, and this should be explicitly documented for anyone who is looking to add similar code in the future.

This is a really important point. There is some scarring here from code history, and it would be sensible to clean up and document.

Before this project began, there was no `PartnersClient` either. We simply had `dev/fetch.ts` which contained both the Partners calls and the logic around them, plus some other junk that collected. And it made sense to test everything in a consistent way. So we treated the functions as units of work, and tested them whether they were just wrappers for Partners or did much more.

I think when we reorganized the code for this project, it created some confusion around the roles of `PartnersClient` and `dev/fetch.ts`. We have moved over most of the code to `PartnersClient` but a few functions are still hanging on in `dev/fetch.ts`, such as `fetchAppDetailsFromApiKey`. It would be wise for us to finish the work and fully separate API wrapping from logic. And along the way, we can reevaluate the tests, add anything missing, and remove anything that isn't creating value.

Reply:
> (Unrelated but I think this should be removed cause the corresponding function this is testing doesn't seem to be used anywhere any more)

This is very weird. Since @shauns added `knip` we shouldn't have unused code. I do see that it's used a couple of times in tests, but not in ways that should count as "real" use.

So, for Shaun:

1. Is this expected?
2. If not, should there be some attempt to tweak knip settings to ensure these kinds of cases are caught and eliminated?
3. If it's expected, should we just manage our expectations for knip and be more active about deleting code?

Reply:
I can't quite follow what the thing is that should be removed, but can talk about knip etc. a little if that's helpful

- ESLint checks for unused values within modules
- Knip checks this stuff: https://knip.dev/reference/issue-types -- which is mainly about dependencies and exports
- We have Knip configured in its default fashion: if something is used in tests, this counts as valid usage and the thing should still be `export`-ed etc.
- This can be turned off with production mode https://knip.dev/features/production-mode -- configured correctly, you can say that test usage doesn't count as _real_ usage
- We don't use this right now. It's (1) probably not correctly configured and (2) likely to be quite disruptive to introduce when its first activated and (3) at the time of introducing Knip we didn't use a version of Vitest that supported in-source testing https://vitest.dev/guide/in-source 
  - Using in-source testing, we could keep tests that are basically checking implementation details, without needing to export those functions
  - Less exporting is good

I would be up for switching to production mode, or at least seeing what it picks up.

Reply:
> It would be wise for us to finish the work and fully separate API wrapping from logic. And along the way, we can reevaluate the tests, add anything missing, and remove anything that isn't creating value.


I've removed the tests for now, I think it sounds like we will need to follow up on both clean up for this old `fetch` file, as well as align on how we want to test API calls in the CLI. What do you think is the best way to ensure this gets done? I was thinking of creating an issue in our team board but I'm nervous this might get bogged down by other issues.